posydon/popsyn/binarypopulation.py

```diff
@@ -520,7 +522,19 @@ def combine_saved_files(self, absolute_filepath, file_names, **kwargs):
                                  min_itemsize=history_min_itemsize)
                     
                     oneline = pd.read_hdf(f, key='oneline')
-                    simulated_mass += oneline['S1_mass_i'].sum() + oneline['S2_mass_i'].sum()
+                    if "initially_single_star" in oneline["state_i"].values:
+                        mask= oneline["state_i"]=="initially_single_star"
+                        filtered_data_single = oneline[mask]
+                        filtered_data_binaries = oneline[~mask]
+    
+                        simulated_mass_binaries = np.sum(filtered_data_binaries[["S1_mass_i", "S2_mass_i"]].to_numpy())
+                        simulated_mass_single = np.sum(filtered_data_single[["S1_mass_i"]].to_numpy())
+                        simulated_mass = simulated_mass_binaries + simulated_mass_single
```

Comment:
The simulated mass should be a sum over several files (as before), which are looped here. (The same applied to the else case.)